tasks/lzards-backup/tests/test-index.js

```diff
@@ -0,0 +1,693 @@
+  const accessUrl = 'fakeURL';
+  const generateAccessUrlMethod = (() => accessUrl);
+  const lzardsPostMethod = (async () => ({
+    body: 'fake body',
+    statusCode: 201,
+  }));
+
+  const roleCreds = { fake: 'creds_object' };
+  const name = 'fakeFilename';
+  const filepath = 'fakeFilePath';
+  const authToken = 'fakeToken';
+  const collectionId = 'FAKE_COLLECTION';
+  const bucket = 'fakeFileBucket';
+  const filename = 's3://fakeFileBucket/fakeFileBucket';
+
+  const file = {
+    name,
+    filepath,
+    bucket,
+    filename,
+  };
+  const granuleId = 'fakeGranuleId';
+
+  const actual = await index.makeBackupFileRequest({
+    authToken,
+    collectionId,
+    roleCreds,
+    file,
+    granuleId,
+    generateAccessUrlMethod,
+    lzardsPostMethod,
+  });
+
+  const expected = {
+    body: 'fake body',
+    filename,
+    granuleId: 'fakeGranuleId',
+    status: 'COMPLETED',
+    statusCode: 201,
+  };
+
+  t.deepEqual(actual, expected);
+});
+
+test('getGranuleCollection returns queried collections', async (t) => {
+  const collectionName = 'fakeCollection';
+  const collectionVersion = '001';
+  const stackPrefix = 'fakePrefix';
+  const actual = await index.getGranuleCollection({
+    collectionName,
+    collectionVersion,
+    stackPrefix,
+  });
+  t.deepEqual(actual, fakeCollection);
+});
+
+test('getGranuleCollection throws error if no prefix is set', async (t) => {
+  const collectionName = 'fakeCollection';
+  const collectionVersion = '001';
+  await t.throwsAsync(index.getGranuleCollection({
+    collectionName,
+    collectionVersion,
+  }));
+});
+
+test('getGranuleCollection throws error if version and name are not defined', async (t) => {
+  const stackPrefix = 'fakePrefix';
+  await t.throwsAsync(index.getGranuleCollection({
+    stackPrefix,
+  }));
+});
+
+test.serial('postRequestToLzards creates the expected query', async (t) => {
+  const accessUrl = 'fakeUrl';
+  const authToken = 'fakeToken';
+  const collection = 'fakeCollectionString';
+  const file = { fake: 'fileObject', filename: 'fakeFilename', checksumType: 'md5', checksum: 'fakeChecksum' };
+  const granuleId = 'fakeGranuleId';
+  const lzardsApi = 'fakeApi';
+  const lzardsProviderName = 'fakeProvider';
+
+  process.env.lzards_provider = lzardsProviderName;
+  process.env.lzards_api = lzardsApi;
+
+  const actual = await index.postRequestToLzards({
+    accessUrl,
+    authToken,
+    collection,
+    file,
+    granuleId,
+    lzardsApi,
+    lzardsProviderName,
+  });
+
+  t.is(actual, fakePostReturn);
+  t.deepEqual(gotPostStub.getCalls()[0].args, [lzardsApi, {
+    json: {
+      provider: lzardsProviderName,
+      objectUrl: accessUrl,
+      expectedMd5Hash: file.checksum,
+      metadata: {
+        filename: file.filename,
+        collection,
+        granuleId,
+      },
+    },
+    headers: {
+      Authorization: `Bearer ${authToken}`,
+      'Content-Type': 'application/json',
```

Comment:
We shouldn't be specifying this header anymore, so I would think that this is failing?

Reply:
Correct you are: 

d46757fe5